### PR TITLE
Prefer InvokeVirtual in tests, avoid passing constants

### PR DESCRIPTION
**Pull request by som-snytt**
Revert `itf = false` and instead prefer special method.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala [hunk 1]
```diff
@@ -1150,6 +1150,10 @@ object BTypes {
       unsafeWrapArray(result)
     }
   }
+  object InlineInfo {
+    final def NOT_FINAL = false
+    final def NOT_SAM = None
+  }
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala]: not used enough to pull its weight

Comment (joroKr21) [reply src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala]: I tend to agree 👍

File: test/junit/scala/tools/nsc/backend/jvm/opt/ScalaInlineInfoTest.scala [hunk 2]
```diff
@@ -83,35 +82,35 @@ class ScalaInlineInfoTest extends BytecodeTesting {
     val cs @ List(c, t, tl, to) = compileClasses(code)
     val infoT = inlineInfo(t)
     val expectT = InlineInfo (
-      false, // final class
-      None, // not a sam
+      NOT_FINAL,
+      NOT_SAM,
```
Comment (som-snytt) [inline test/junit/scala/tools/nsc/backend/jvm/opt/ScalaInlineInfoTest.scala]: `EmptyInlineInfo.copy(methodInfos = mystuff)`

**Comment by som-snytt** [comment]
I like the "empty prototype" idiom (`EmptyInlineInfo.copy`) better than default args for constructor (where defaults are zero values).

**Review by joroKr21 — APPROVED** [review]
I'm a sucker for helper methods 👍

**Comment by som-snytt** [comment]
I wonder if @SethTisue will take a break from holiday carousing (or work) to see what ruby issue derails the travis build.

**Review by lrytz — APPROVED** [review]
👌

**Pull request merged**